truelearn/datasets/_base.py

```diff
@@ -0,0 +1,105 @@
+import dataclasses
+import hashlib
+from typing import Optional
+from urllib import request
+from os import path
+
+
+# pylint: disable=pointless-string-statement
+"""
+Copyright of `RemoteFileMetaData`, `download_file` are held by
+[BSD 3-Clause License, scikit-learn developers, 2007-2022].
+"""
+
+
+@dataclasses.dataclass
+class RemoteFileMetaData:
+    """Remote file metadata.
+
+    Args:
+        url: The url of the file.
+        filename: The filename of the downloaded file.
+        sha256_expected: The expected sha256 sum of the file.
```

Comment:
Unresolved ref sha256_expected, as you called it sha256 on line 27, maybe expected_sha256 is better name

Reply:
I will fix that later.